astroid/objects.py

```diff
@@ -266,11 +266,15 @@ class DictValues(bases.Proxy):
 class PartialFunction(scoped_nodes.FunctionDef):
     """A class representing partial function obtained via functools.partial"""
 
+    @decorators.deprecate_arguments(doc="Use the postinit arg 'doc_node' instead")
     def __init__(
         self, call, name=None, doc=None, lineno=None, col_offset=None, parent=None
     ):
         # TODO: Pass end_lineno and end_col_offset as well
-        super().__init__(name, doc, lineno, col_offset, parent=None)
+        super().__init__(name, lineno=lineno, col_offset=col_offset, parent=None)
+        # TODO: Set directly in order not to trigger DeprecationWarning. Remove
+        # when '.doc' is removed.
```

Comment:
```suggestion
        # Assigned directly to prevent triggering the DeprecationWarning.
```
Tbh, we don't even need a `TODO` here. Once the `doc` argument is removed, any linter will notice that.